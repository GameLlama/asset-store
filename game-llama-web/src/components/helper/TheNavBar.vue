<template>
  <header>
    <v-toolbar>
      <!-- <v-toolbar-side-icon></v-toolbar-side-icon> -->
      <v-layout row wrap>
        <v-flex xs4>
          <router-link :to="{name:'Home'}" class="logo">
            <img src="../../assets/logo_header.png">
          </router-link>
        </v-flex>
        <v-flex xs4>
          <v-text-field
              label="Search in Game Llama"
              prepend-inner-icon="search"
              class="search"
              clearable>
          </v-text-field>
        </v-flex>
        <v-flex xs4>
          <div class="right-icons">
            <v-btn flat icon color="#263959">
              <v-icon large>edit</v-icon>
            </v-btn>
            <v-badge  color="red" overlap>
              <span   v-if="messageCounter" slot="badge">0</span>
              <v-btn flat icon color="#263959">
                <v-icon large>email</v-icon>
              </v-btn>
            </v-badge>
            <v-menu bottom left>
              <v-btn dark icon color="#263959" slot="activator">
                <v-icon large>account_circle</v-icon>
              </v-btn>
              <v-list>
                <v-list-tile v-for="item in items" :key="item.key">
                  <v-list-tile-title>{{ item.title }}</v-list-tile-title>
                </v-list-tile>
              </v-list>
            </v-menu>
          </div>
        </v-flex>
      </v-layout>
    </v-toolbar>
  </header>
</template>

<script>
export default {
  data() {
    return {
      messageCounter: false,
      profileCounter: false,
      items: [
        { title: 'Click Me' },
        { title: 'Click Me' },
        { title: 'Click Me' },
        { title: 'Click Me 2' },
      ],
    };
  },
};
</script>
