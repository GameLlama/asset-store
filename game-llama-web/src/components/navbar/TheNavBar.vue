<template>
  <header>
    <v-toolbar>
      <v-layout row wrap>
        <v-flex xs3>
          <router-link :to="{name:'Home'}" class="logo">
            <img src="../../assets/logo_header.png">
          </router-link>
        </v-flex>
        <v-flex xs6>
          <search-bar class="search"></search-bar>
        </v-flex>
        <v-flex xs3>
          <div class="right-icons">
            <v-btn flat icon color="#263959">
              <v-icon large>add_circle</v-icon>
            </v-btn>
            <v-badge  color="red" overlap>
              <span   v-if="messageCounter" slot="badge">0</span>
              <v-btn flat icon color="#263959">
                <v-icon large>notification_important</v-icon>
              </v-btn>
            </v-badge>
            <user-menu></user-menu>
          </div>
        </v-flex>
      </v-layout>
    </v-toolbar>
  </header>
</template>

<script>

import UserMenu from '@/components/navbar/UserMenu';
import SearchBar from '@/components/navbar/SearchBar';

export default {
  components: {
    'user-menu': UserMenu,
    'search-bar': SearchBar,
  },
  data() {
    return {
      messageCounter: false,
      profileCounter: false,
    };
  },
};
</script>
