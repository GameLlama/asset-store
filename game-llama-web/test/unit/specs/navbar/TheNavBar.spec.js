import Vue from 'vue';
import TheNavBar from '@/components/navbar/TheNavBar';

describe('TheNavBar.vue', () => {
  let vm;
  let Component;
  beforeEach(() => {
    Component = Vue.extend(TheNavBar);
    vm = new Component().$mount();
  });

  it('has the expected structure', () => {
    expect(vm.$el).toMatchSnapshot();
  });
});
