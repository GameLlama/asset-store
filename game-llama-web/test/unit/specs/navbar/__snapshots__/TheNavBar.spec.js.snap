// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheNavBar.vue has the expected structure 1`] = `
<header>
  <v-toolbar>
    <v-layout
      row=""
      wrap=""
    >
      <v-flex
        xs3=""
      >
        <router-link
          class="logo"
          to="[object Object]"
        >
          <img
            src="../../assets/logo_header.png"
          />
        </router-link>
      </v-flex>
       
      <v-flex
        xs6=""
      >
        <!---->
      </v-flex>
       
      <v-flex
        xs3=""
      >
        <div
          class="right-icons"
        >
          <v-btn
            color="#263959"
            flat=""
            icon=""
          >
            <v-icon
              large=""
            >
              add_circle
            </v-icon>
          </v-btn>
           
          <!---->
           
          <!---->
        </div>
      </v-flex>
    </v-layout>
  </v-toolbar>
</header>
`;
